src/socket.io/index.js

```diff
@@ -191,14 +197,18 @@ const getSessionAsync = util.promisify(
 	(sid, callback) => db.sessionStore.get(sid, (err, sessionObj) => callback(err, sessionObj || null))
 );
 
-async function validateSession(socket) {
+async function validateSession(socket, isRevalidate) {
 	const req = socket.request;
 	if (!req.signedCookies || !req.signedCookies[nconf.get('sessionKey')]) {
 		return;
 	}
 	const sessionData = await getSessionAsync(req.signedCookies[nconf.get('sessionKey')]);
 	if (!sessionData) {
-		throw new Error('[[error:invalid-session]]');
```

Comment:
Instead of using a boolean flag you can pass in the error message you want thrown. Then it would be simply.
```
async function validateSession(socket, errorMsg) {
...
  if (!sessionData) {
 	throw new Error(errorMsg);
  }
...
}
```

Reply:
:100: